### ISSUE TITLE
Unrecognized Redshift function SPLIT_PART

### ISSUE THREAD

**Issue by anikolaienko** [sqlfluff/sqlfluff#3710]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

My problem is that `SPLIT_PART` keyword is not recognized for redshift dialect. Here is the function:
https://docs.aws.amazon.com/redshift/latest/dg/SPLIT_PART.html

But my actual suggestion would be to have ability to add keywords in config for quick testing and use on production and after try to apply this to main code.

### Expected Behaviour

Ability to add keywords in config

### Observed Behaviour

I could not find how to quickly fix my issue. Also `-- ignore` does not work either.

### How to reproduce

Any example of using SPLIT_PART: https://docs.aws.amazon.com/redshift/latest/dg/SPLIT_PART.html

### Dialect

Redshift

### Version

1.2.1

### Configuration

[sqlfluff]
verbose = 1
dialect = redshift
templater = jinja
exclude_rules = L003, L033, L031, L042, L066

[sqlfluff:rules]
tab_space_size = 2
max_line_length = 300

[sqlfluff:templater:jinja]
library_path = sqlfluff_libs

### Are you willing to work on and submit a PR to address the issue?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by anikolaienko** [comment]
I am also trying to understand is there a way to ignore these types of errors:
```
  [1 unfixable linting violations found]
```
So it would be processed as a warning rather than failing check. I am using sqlfluff-fix with pre-commit

**Comment by barrywhart** [comment]
Re: the "unfixable" error message, the next release of SQLFluff replaces this message with a list of the unfixable issues. This may help you.

**Comment by pdebelak** [comment]
This function is also part of [postgres](https://www.postgresql.org/docs/current/functions-string.html#id-1.5.8.10.7.2.2.31.1.1.1) so we should ensure its recognized in both `postgres` and `redshift` dialects.

**Comment by anikolaienko** [comment]
Thank you guys!
Is there a quick fix without new release that I can apply either to my SQL code or config?
Some sort of `--ignore` comment in SQL or provide an additional keyword list to be recognised in config?

**Comment by pdebelak** [comment]
@anikolaienko The ignore comment is `--noqa`. See https://docs.sqlfluff.com/en/stable/rules.html#inline-ignoring-errors

**Comment by anikolaienko** [comment]
Sorry guys, I think my problem is not with `SPLIT_PART`, although removing this line actually fixed the issue.

The actual testcase which produces `[1 unfixable linting violations found]` is:
```sql
SELECT
    SPLIT_PART(field, ',', 2) AS title,  -- also reproduced with any other function (e.g. TRIM)
    content                                           -- important that that is a field after line where function is used
FROM test_table
GROUP BY 1                                       -- important that this GROUP BY works by indexes, not field names
```
I also figured that `--noqa` only works when I place it on `SELECT` statement line.

**Comment by pdebelak** [comment]
Yes, this is the behavior of rule L034, it doesn't fix if you are using implicit column references at the moment. See https://github.com/sqlfluff/sqlfluff/blob/main/src/sqlfluff/rules/L034.py#L160-L166

**Comment by WittierDinosaur** [comment]
To confirm, redshift is currently recognising this as a function. This is expected L034 behaviour, so you either need to disable the rule, or change the style of your query.

**Comment by anikolaienko** [comment]
You are right. I solved it by disabling the rule. I will close this request.
